addon/-private/functional/modifier.ts

```diff
@@ -103,12 +163,12 @@ export default function modifier(
   // returns an opaque `Modifier` type so that we can provide a result from this
   // type which is useful to TS-aware tooling (e.g. Glint).
   return setModifierManager(
-    () => FunctionalModifierManager,
+    () => new FunctionalModifierManager(options),
```

Comment:
Instead of making a new instance per `modifier(` call, let's just make two: one for eager and one for lazy; then hand back those singeltons...

Reply:
Ahhh, thank you! This was the one thing I was thinking about.